pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/impl/LeastResourceUsageWithWeight.java

```diff
@@ -125,6 +125,11 @@ private double updateAndGetMaxResourceUsageWithWeight(String broker, BrokerData
     @Override
     public Optional<String> selectBroker(Set<String> candidates, BundleData bundleToAssign, LoadData loadData,
```

Comment:
it can not permit after invoke `applyNamespacePolicies`  candidates is not empty. there may be not any one broker satisfy the NamespacePolicies for the namespace.

Reply:
It makes sense. LGTM